include/fmt/core.h

```diff
@@ -252,6 +252,21 @@
 #  pragma execution_character_set("utf-8")
 #endif
 
+#if defined(FMT_ENABLE_COMPILE_TIME_FORMATTING)
+#  if !defined(FMT_HEADER_ONLY) || __cplusplus < 202002L
+#    error "compile time formatting could not be enabled"
+#  endif
+#  define FMT_CTF_CONSTEXPR constexpr
+#  define FMT_CTF_CUSTOM_BACK_INSERT_ITERATOR 1
+#  define FMT_CTF_CHECK std::is_constant_evaluated()
+#  define FMT_CTF_ENABLED 1
+#else
+#  define FMT_CTF_CONSTEXPR
+#  define FMT_CTF_CUSTOM_BACK_INSERT_ITERATOR 0
+#  define FMT_CTF_CHECK false
+#  define FMT_CTF_ENABLED 0
+#endif
```

Comment:
This shouldn't depend on `FMT_ENABLE_COMPILE_TIME_FORMATTING`. Instead use the normal {fmt} macros such as `FMT_CONSTEXPR` adding standard-specific variants if necessary, e.g.

```c++
#ifdef __cplusplus >= 201703L
#  define FMT_CONSTEXPR17 FMT_CONSTEXPR
#endif
```

Also I suggest moving constexprification to a separate PR since it's a pretty mechanical change.

Reply:
Sounds like a lot of work, but I agree that I should be done this way. In the PR I simplify some things to get feedback for other things.